app/src/pages/inside/settingsPage/settingsTabs/autoAnalysisTab/analysisForm/analysisForm.jsx

```diff
@@ -0,0 +1,233 @@
+  },
+  Light: {
+    minShouldMatch: 60,
+    minDocFreq: 7,
+    minTermFreq: 1,
+    numberOfLogLines: 2,
+  },
+};
+
+const DEFAULT_ANALYSIS_MODE = 'Moderate';
+
+@injectIntl
+@reduxForm({
+  form: 'analysisForm',
+  validate: ({ minShouldMatch, minTermFreq, minDocFreq }) => ({
+    minShouldMatch:
+      (!minShouldMatch || !validate.minShouldMatch(minShouldMatch)) && 'minShouldMatchHint',
+    minTermFreq: (!minTermFreq || !validate.minTermFreq(minTermFreq)) && 'minTermFreqHint',
+    minDocFreq: (!minDocFreq || !validate.minDocFreq(minDocFreq)) && 'minTermFreqHint',
+  }),
+})
+@connect(
+  (state) => ({
+    analyzerConfiguration: projectAnalyzerConfigSelector(state),
+    currentProject: activeProjectSelector(state),
+    formInputsValues: selector(
+      state,
+      'minShouldMatch',
+      'minDocFreq',
+      'minTermFreq',
+      'numberOfLogLines',
+    ),
+  }),
+  { showNotification },
+)
+export class AnalysisForm extends PureComponent {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    analyzerConfiguration: PropTypes.object,
+    currentProject: PropTypes.string,
+    initialize: PropTypes.func,
+    change: PropTypes.func,
+    handleSubmit: PropTypes.func,
+    showNotification: PropTypes.func,
+    formInputsValues: PropTypes.object,
+  };
+
+  static defaultProps = {
+    analyzerConfiguration: {},
+    currentProject: '',
+    initialize: () => {},
+    change: () => {},
+    handleSubmit: () => {},
+    showNotification: () => {},
+    formInputsValues: {},
+  };
+
+  state = {
+    currentTabsValue: DEFAULT_ANALYSIS_MODE,
+  };
+
+  componentDidMount() {
+    const initialConfiguration = {
+      ...this.props.analyzerConfiguration,
+    };
+    delete initialConfiguration.indexing_running;
+    this.setAnalysisMode(initialConfiguration);
+    this.props.initialize(initialConfiguration);
+  }
+
+  onInputChange = (event, newValue, previousValue, name) => {
+    const formattedValue = newValue.toString().replace(/\D+/g, '');
+    const formInputsValues = this.props.formInputsValues;
+    formInputsValues[name] = +formattedValue;
+    this.setAnalysisMode(formInputsValues);
+  };
+
+  onFormSubmit = (data) => {
+    const dataToSend = {
+      configuration: {
+        analyzerConfiguration: {
+          ...data,
+        },
+      },
+    };
+    fetch(URLS.project(this.props.currentProject), { method: 'put', data: dataToSend })
+      .then(() => {
+        this.props.showNotification({
+          message: this.props.intl.formatMessage(messages.updateSuccessNotification),
+          type: 'success',
+        });
+      })
+      .catch(() => {
+        this.props.showNotification({
+          message: this.props.intl.formatMessage(messages.updateErrorNotification),
+          type: 'error',
+        });
+      });
+  };
+
+  setAnalysisMode = (modeConfig) => {
+    let isModeChosen = false;
+
+    const analysisModeKeys = Object.keys(analysisModeConfig);
+    analysisModeKeys.forEach((key) => {
```

Comment:
I think `find` looks more suitable here since you want to find only one value

Reply:
Fixed